modules/hub/src/ChainsawService.ts

```diff
@@ -272,10 +272,60 @@ export default class ChainsawService {
     if (!disputeRecord) {
       // dispute might not have been initiated by us, so we need to add it here
       await this.channelDisputesDao.create(event.user, 'Dispute caught by chainsaw', chainsawId, null, onchainChannel.channelClosingTime)
-      return
+    } else {
+      await this.channelDisputesDao.setExitEvent(disputeRecord.id, chainsawId, onchainChannel.channelClosingTime)
```

Comment:
I'm not 100% certain, but it seems like this should be:

```
if (!disputeRecord)
  disputeRecord = this.channelDipsutesDao.create(…)
await this.channelDisputesDao.setExitEvent(…)
```

Reply:
Absolutely correct.

Reply:
Should there still be the `} else { this.channelDisputesDao.setExitEvent(…) }` here?

Reply:
Aren't those the same thing?

Reply:
@ArjunBhuptani, no, one is creating the record in the db, one is modifying an existing record in the db.